pkg/grpc/ratelimit/ratelimit_impl.go

```diff
@@ -0,0 +1,106 @@
+package ratelimit
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/ratelimit"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/sync"
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc"
+)
+
+type rateLimiter struct {
+	mutex              sync.Mutex
+	tokenBucketLimiter *rate.Limiter
+}
+
+// Limit implements "ratelimit.Limiter" interface.
+func (limiter *rateLimiter) Limit() bool {
+	return !limiter.tokenBucketLimiter.Allow()
+}
+
+func (limiter *rateLimiter) IncreaseLimit(limit int) {
+	if limiter.tokenBucketLimiter.Limit() == rate.Inf || limit <= 0 {
+		return
+	}
+
+	limiter.mutex.Lock()
+	defer limiter.mutex.Unlock()
+
+	newBurst := limiter.tokenBucketLimiter.Burst() + limit
+	if 0 < newBurst {
```

Comment:
Looks like it's always true

Reply:
It is an integer addition overflow guard. If a negative burst or `0` is set, the rate limiter would block all events.